library/osrm_impl.hpp

```diff
@@ -28,6 +28,10 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifndef OSRM_IMPL_HPP
 #define OSRM_IMPL_HPP
 
+// #if __cplusplus > 199711L
```

Comment:
I guess it's because @chaupow has a old boost version installed globally and recent compiler versions (rightfully) warn for the deprecation of the register keyword in C++11.

@chaupow can you upgrade your boost installation? Or install a recent version locally, e.g. in your home directory and then tell CMake the path to this directory.